lib/puppet/util/nc_https.rb

```diff
@@ -155,6 +155,16 @@ def get_environments
     JSON.parse(res.body)
   end
 
+  def get_nodes(name)
+    res = do_https("v1/nodes/#{name}", 'GET')
```

Comment:
To provider as much coverage on top of the classifier API, can you do it so that if no argument is specified, it will hit `/v1/nodes`?  Maybe something like:
```
def get_nodes(name = nil)
  endpoint = ['v1/nodes', name].compact.join('/')
  res = do_https(endpoint, 'GET')
  ...
end
```

Reply:
@mhjacks

Reply:
I've added the ability to call the method without arguments, and documented it it readme.  The PE documentation notes that calling the endpoint without arguments is deprecated, and I've noted that the README as well.

I've left the method name as get_nodes() for the moment - the other method name all have {http_verb}_{endpoint} naming convention.  I'm happy to change that if you really want me to, but I thought it would be more convenient for users/consumers to maintain the naming convention already in place in that module.  What do you think @WhatsARanjit ?